test/python/transpiler/test_mappers.py

```diff
@@ -0,0 +1,166 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""Meta Tests for mappers
+
+Regenerate ground truth by running this on the root directory:
+> python -m  test.python.transpiler.test_mappers
+
+"""
+
+# pylint: disable=redefined-builtin,no-member,attribute-defined-outside-init
+
+import unittest
+import pickle
+
+from qiskit import ClassicalRegister, QuantumRegister, QuantumCircuit, BasicAer, compile
+from qiskit.transpiler import PassManager, transpile
+from qiskit.transpiler.passes import BasicSwap, LookaheadSwap, StochasticSwap
+from qiskit.mapper import CouplingMap
+
+from ..common import QiskitTestCase
+
+
+class CommonUtilities():
+    """ Some utilities for meta testing."""
+    regenerate_expected = False
+    seed = 42
+    additional_args = {}
+
+    def create_passmanager(self, coupling_map):
+        ''' Returns a PassManager using self.pass_class and coupling_map '''
+        return PassManager(self.pass_class(CouplingMap(coupling_map), **self.additional_args))
+
+    def create_backend(self):
+        ''' Returns a Backend.'''
+        return BasicAer.get_backend('qasm_simulator')
+
+    def generate_expected(self, result, filename):
+        """
+        Checks if result.get_count matches self.count by running in a backend
+        (self.create_backend()). That's saved in a pickle in filename.
+
+        Args:
+            result (DAGCircuit): The DAGCircuit to compile and run.
+            filename (string): Where the pickle is saved.
+        """
+        sim_backend = self.create_backend()
+        qobj = compile(result, sim_backend, seed=self.seed, shots=self.shots)
+        job = sim_backend.run(qobj)
+        self.assertDictAlmostEqual(self.count, job.result().get_counts(), delta=self.delta)
+
+        with open(filename, "wb") as output_file:
+            pickle.dump(result, output_file)
+
+    def assertResult(self, result, testname):
+        ''' Fetches the pickle in testname file and compares it with result'''
+        filename = self._get_resource_path('pickles/%s_%s.pickle' % (type(self).__name__, testname))
```

Comment:
how does this `type(self)` become the name of different Swap pass classes? I'm sure it works but I'm just not following this part.

Reply:
`type(self)` is `TestsXSwap` not the swap pass class.